File: utils/DateManager.py
from datetime import datetime
from exception.InternalServerErrorException import InternalServerErrorException

class DateManager:
    DAYS = {
        0: 'Lunes',
        1: 'Martes',
        2: 'Miércoles',
        3: 'Jueves',
        4: 'Viernes',
        5: 'Sábado',
        6: 'Domingo'
    }

    @staticmethod
    def GetDayNameFromDate(date_str):
        try:

            date_obj =  datetime.strptime(date_str, "%Y-%m-%dT%H:%M:%S%z").date()
            day_week = date_obj.weekday() 
            return DateManager.DAYS.get(day_week, "Día inválido") 

        except Exception as e:
            raise InternalServerErrorException(error=str(e))
    @staticmethod
    def GetFullDateInText(date_str):
        """
        Retorna la fecha en el formato: 'DD/MM/YYYY HH:MM AM/PM'
        """
        try:
            # Convertir string a datetime
            date_obj = datetime.strptime(date_str, "%Y-%m-%dT%H:%M:%S%z").date()


            # Formatear la fecha en 'DD/MM/YYYY HH:MM AM/PM'
            formatted_date = date_obj.strftime("%d/%m/%Y %I:%M %p")

            return formatted_date

        except Exception as e:
            raise InternalServerErrorException(error=str(e))

File: app/Utils/DateManager.php
<?php
namespace App\Utils;

use Carbon\Carbon;
use App\Exceptions\CustomExceptions\ServerErrorException;

class DateManager {

    static public $daysWeek = [
        0 => 'Domingo',
        1 => 'Lunes',
        2 => 'Martes',
        3 => 'Miércoles',
        4 => 'Jueves',
        5 => 'Viernes',
        6 => 'Sábado'
    ];

    /**
     * Devuelve la fecha en formato corto: "Día/Mes Hora AM/PM"
     * 
     * @param mixed $date (string, DateTime o Carbon)
     * @return string
     * @throws ServerErrorException Si la fecha no es válida
     */
    public static function getDshrtDate($date) {
        try {
            $date = self::parseDate($date);
            return $date->format('d/m h:i A');
        } catch (\Exception $e) {
            throw new ServerErrorException("Fecha inválida proporcionada.");
        }
    }

    /**
     * Devuelve el nombre del día de la semana para una fecha dada.
     * 
     * @param mixed $date (string, DateTime o Carbon)
     * @return string
     * @throws ServerErrorException Si la fecha no es válida
     */
    public static function getDayWeekToDate($date) {
        try {
            $date = self::parseDate($date);
            return self::$daysWeek[$date->dayOfWeek];
        } catch (\Exception $e) {
            throw new ServerErrorException("Fecha inválida proporcionada.");
        }
    }

    /**
     * Convierte una fecha en un objeto Carbon, sin importar el tipo de entrada.
     * 
     * @param mixed $date (string, DateTime o Carbon)
     * @return Carbon
     * @throws \Exception Si la fecha no es válida
     */
    private static function parseDate($date) {
        if (!$date instanceof Carbon) {
            return Carbon::parse($date);
        }
        return $date;
    }
}
